src/pages/fi-tax-schedules/forms/TaxSchedulesForm.js

```diff
@@ -0,0 +1,125 @@
+import { Grid, FormControlLabel, Checkbox } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useForm } from 'src/hooks/form'
+import { FinancialRepository } from 'src/repositories/FinancialRepository'
+import { MasterSource } from 'src/resources/MasterSource'
+
+export default function TaxSchedulesForm({ labels, maxAccess, setStore, store }) {
+  const { recordId } = store
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: FinancialRepository.TaxSchedules.qry
+  })
+
+  const { formik } = useForm({
+    initialValues: { recordId: null, name: '', reference: '', nonDeductible: false },
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      name: yup.string().required(' '),
+      reference: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+
+      const response = await postRequest({
+        extension: FinancialRepository.TaxSchedules.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        setStore(prevStore => ({
+          ...prevStore,
+          recordId: response.recordId
+        }))
+        toast.success('Record Added Successfully')
+        formik.setValues({
+          ...obj,
+          recordId: response.recordId
+        })
+      } else toast.success('Record Edited Successfully')
+
+      invalidate()
+    }
+  })
+  const editMode = !!formik.values.recordId || !!recordId
```

Comment:
We're already setting the store recordId. we can stick to !!recordId. I guess editMode is passed from the parent too.